### PR TITLE
Reimplement NVN

### PR DESCRIPTION
**Pull request by ThePixelGamer**
This PR uses dwarf info to more closely match the original file layout.
There's still a few more things that need to be done for complete-ness sake

- [x] Implement the cpp version found in Diablo 3's dwarf build
- [ ] Implement a versioning system to remove functions found in the more recent nvn used in the Darksiders 2 dwarf build
- [ ] Verify the file structure, primarily using the cpp's files as reference (ie: nvn_CppFuncPtr*.h, no references to a *Base.h but it's needed for *Impl.h)
- [ ] Remove nvnInit.cpp (can't be done in this PR, must be done in the nn::gfx PR)

This PR also breaks the agl library build and botw repo build so follow up PRs will be needed

- - -
This change is [Reviewable](https://reviewable.io/reviews/open-ead/nnheaders/14)

### CODE REVIEW

**Comment by leoetlino** [comment]
I'm confused by the FuncPtrMacro / FuncPtrInline stuff. Don't they conflict with each other?

**Comment by ThePixelGamer** [comment]
Yes but I figured I would leave them for compat with the older systsmy

**Comment by ThePixelGamer** [comment]
The other 2 (ignoring nvnInit) could be done in this PR but it might be too much, should be good to merge now

File: include/nvn/nvn_Cpp.h [hunk 1]
```diff
@@ -3,47 +3,97 @@
 #include <stddef.h>
 #include <stdint.h>
 
-// todo: implement?
+// todo: figure out m_value gets set to by the empty constructor
```
Comment (leoetlino) [inline include/nvn/nvn_Cpp.h:6]:
```suggestion
// todo: figure out what m_value gets set to by the empty constructor
```

**Comment by leoetlino** [comment]
Other than the comment typo, is this good to go?

**Comment by ThePixelGamer** [comment]
Yes

**Pull request merged**